### PR TITLE
Cypress component test

### PR DESCRIPTION
**Pull request by avanti-em**

### CODE REVIEW

File: .github/workflows/webviz-subsurface-components.yml [hunk 1]
```diff
@@ -102,24 +102,50 @@ jobs:
...
+          cd react
+          npx cypress install
+
+      # run the cypress tests
+      - name: e2e on firefox
+        if: github.event_name != 'release' && matrix.e2e-tests
+        run: |
+          mkdir clone
+          cd clone
+          git clone --branch cypress-snaps --depth 1 --filter=blob:none https://github.com/avanti-em/webviz-subsurface-components
```
Comment (anders-kiaer) [inline .github/workflows/webviz-subsurface-components.yml:129]: Should this be changed to official repo before merge?

File: .github/workflows/webviz-subsurface-components.yml [hunk 2]
```diff
@@ -155,7 +181,7 @@ jobs:
         run: npm run build-storybook --prefix ./react
 
       - name: 📚 Update Storybook and GitHub demo application
-        if: github.event_name == 'push' && (github.ref == 'refs/heads/master' || contains(github.event.head_commit.message, '[deploy test]')) && matrix.python-version == '3.8'
+        if: github.event_name == 'push' && (github.ref == 'refs/heads/master' || contains(github.event.head_commit.message, '[deploy test]')) && matrix.python-version == '3.8' && failure() && matrix.e2e-tests
```
Comment (anders-kiaer) [inline .github/workflows/webviz-subsurface-components.yml:184]: For my understanding: Why do we want to update official storybook only on failure?

File: .github/workflows/webviz-subsurface-components.yml [hunk 3]
```diff
@@ -176,3 +202,32 @@ jobs:
...
+        id: cpr
+        uses: peter-evans/create-pull-request@5a6b15373e5788c38d83296b674ad5abea28085c
+        if: steps.check_file_changed.outputs.status == 'true' && github.event_name == 'push' && (github.ref == 'refs/heads/master' || contains(github.event.head_commit.message, '[deploy test]')) && matrix.python-version == '3.10' && failure()
+        with:
+          commit-message: updated cypress snapshots
+          title: Cypress - New Snapshots generated for the commit "${{ github.event.head_commit.message }}" by ${{ github.event.commits[0].author.email }}
+          body: This is an auto-generated PR to update cypress snapshots.
+          base: cypress-snaps
+          reviewers: ${{ github.event.sender.login }}
+          delete-branch: true
```
Comment (anders-kiaer) [inline .github/workflows/webviz-subsurface-components.yml:233]: Missing newline

**Review by anders-kiaer — COMMENTED** [review]
A couple of small questions, otherwise looks great - nice work. 🙂 

As discussed earlier, it is hard to test this fully prior to merge as it is about the CI pipeline itself, with some complex features.

File: .github/workflows/webviz-subsurface-components.yml (hunk 1 above)
Comment (avanti-em) [reply .github/workflows/webviz-subsurface-components.yml:129]: Oops! I forgot to change it. Thanks for catching.

File: .github/workflows/webviz-subsurface-components.yml (hunk 2 above)
Comment (avanti-em) [reply .github/workflows/webviz-subsurface-components.yml:184]: I think I did some kind of mistake while I was doing some "search and replace" stuff. Its a complete mistake, I will undo it.

File: .github/workflows/webviz-subsurface-components.yml (hunk 3 above)
Comment (avanti-em) [reply .github/workflows/webviz-subsurface-components.yml:233]: Got it, will fix it

**Review by anders-kiaer — APPROVED** [review]

**Pull request merged**